flytekit/exceptions/base.py

```diff
@@ -7,6 +10,24 @@ def error_code(cls):
 class FlyteException(Exception, metaclass=_FlyteCodedExceptionMetaclass):
     _ERROR_CODE = "UnknownFlyteException"
 
+    def __init__(self, message: str) -> None:
```

Comment:
I'm personally unsure whether we want to set an exception timestamp on all Flyte exceptions ever raised by default.

Alternatively we could do:

```py
def __init__(self, message: str, timestamp: Optional[...]=None) -> None:
```

Only plugins that really need the timestamp would then set it.

In `get_timestamp` we'd have to check whether `e.timestamp is None`.

Not saying that I have a strong opinion that this is better but this is what I would have done in order to limit the "area of effect".

Reply:
Sure, makes sense.